pkg/systemd/host.js

```diff
@@ -566,6 +576,20 @@ PageServer.prototype = {
             refresh_os_updates_state();
         });
 
+        function refresh_service_failures() {
+            const status = page_status.get("system/services");
+
+            $("#system_information_service_failures").empty();
+            if (status) {
+                $("#system_information_service_failures")
+                        .append(status.details.map(u => $('<div>').append(
+                            arrfmt(_("$0 has failed"), $('<a>', { 'data-goto-service': u }).text(u)))));
```

Comment:
This is the only usage of arrfmt(), and it feels a bit overkill. It could just use `.push(cockpit.format(_("$0 has failed"), ...))` instead?

Reply:
The point of arrfmt is to put things into a formatting string that are not strings, such as React components or jQuery things.

Thus, `arrfmt("$0 has failed", $(<a>))` is the same as `[ $(<a>), " has failed" ]`.  The link will appear in the DOM.

`cockpit.format("$0 has failed", $(<a>))` is the same as " has failed" and is not what we want.

Reply:
Ah, I see. Thanks!